src/components/page/elements/MultipleChoiceQuestionElement.js

```diff
@@ -0,0 +1,255 @@
+    this.setState({ submitted: true, answeredCorrectly, showHints: !answeredCorrectly })
+    console.log('Your answer is ', answeredCorrectly)
+  }
+
+  handleInputChange(event) {
+    const target = event.target
+    const value = target.type === 'checkbox' ? target.checked : target.value
+    const name = target.name
+    if (this.props.content.multiplechoice) {
+      let newSelectedState = this.state.selected
+      if (value && newSelectedState.filter(itm => itm === name).length === 0) {
+        newSelectedState.push(name)
+      }
+      if (!value && newSelectedState.filter(itm => itm === name).length >= 1) {
+        newSelectedState = newSelectedState.filter(itm => itm !== name)
+      }
+      this.setState({ selected: newSelectedState })
+    } else {
+      this.setState({ selected: [value] })
+    }
+  }
+
+  checkAnswers() {
+    let answeredCorrectly = false
+    const correctAnswers = this.getCorrectAnswers()
+    if (!this.props.content.multiplechoice) {
+      answeredCorrectly = correctAnswers.length === 1 && correctAnswers[0] === this.state.selected[0]
+    } else {
+      if (correctAnswers.length === this.state.selected.length) {
+        answeredCorrectly = correctAnswers.reduce((acc, value) => {
+          return acc || this.state.selected.filter(itm => itm === value).length >= 1
+        }, false)
+      }
+    }
+    return answeredCorrectly
+  }
+
+  getCorrectAnswers() {
+    const correctAnswers = []
+    this.props.content.answers.forEach(answer => {
+      if (answer.content[0].value) {
+        if (this.props.content.multiplechoice) {
+          correctAnswers.push(
+            shajs('sha256')
+              .update(JSON.stringify(answer.content[0].answer))
+              .digest('hex'),
+          )
+        } else {
+          if (correctAnswers.length === 0) {
+            const answerHash = shajs('sha256')
+              .update(JSON.stringify(answer.content[0].answer))
+              .digest('hex')
+            correctAnswers.push(answerHash)
+          }
+        }
+      }
+    })
+    return correctAnswers
+  }
+
+  getAnswers() {
+    if (!this.props.content.multiplechoice) {
+      return this.getRadioAnswers()
+    }
+    return this.getCheckboxAnswers()
+  }
+
+  getCheckboxAnswers() {
+    return this.props.content.answers.map(answer => {
+      const answerHash = shajs('sha256')
+        .update(JSON.stringify(answer.content[0].answer))
+        .digest('hex')
+      return (
+        <li>
+          <label className={'checkbox'}>
+            <input
+              type={'checkbox'}
+              className={'mr10'}
+              name={answerHash}
+              value={answer.content[0].answer}
+              checked={this.isSelected(answerHash)}
+              onChange={this.handleInputChange}
+            />
+            <ContentArray content={answer.content[0].answer} />
+          </label>
+        </li>
+      )
+    })
+  }
+
+  getRadioAnswers() {
+    const questionHash = shajs('sha256')
```

Comment:
Use the function described before.

Also, it is unclear to me what `JSON.stringify() + JSON.stringify` looks like. I would rather use `${JSON.stringify}${JSON.stringify}` along with a comment. It can even be its own function since this is core logic

Reply:
fixed